ostis-ai/py-sc-client
# Refactor reconnection

# Reconnection to sc-server

## How to test

Paste this python script

```python
import logging
import signal

from sc_client.client import connect, disconnect

root_logger = logging.getLogger()
root_logger.level = logging.DEBUG
root_logger.addHandler(logging.StreamHandler())

url = "ws://localhost:8090/ws_json"
connect(url, do_reconnect=True, reconnection_time=1.0)

root_logger.info("Press ^C to disconnect")
signal.signal(signal.SIGINT, lambda *_: root_logger.info("^C: disconnecting"))
signal.pause()

disconnect()
```

Now run it. In other terminal run/close sc-server and check logs. 
In any time press ^C (ctrl-C or close using IDE) and client will close connection.
